plugins/base.js

```diff
@@ -13,6 +13,12 @@
 // limitations under the License.
 'use strict';
 
+const debug = require('debug')('cloudant:plugins:base');
+const lockFile = require('lockfile');
+const tmp = require('tmp');
+
+function noop() {}
```

Comment:
I still need this noop func. If we error getting the lock then we still proceed in fetching the session cookie. This means the client is always going to call the callback.
If we success in acquiring the lock then the callback is releasing the lock, otherwise it's this noop func.